github_issue_summarization/estimator/seq2seq_utils.py

```diff
@@ -0,0 +1,432 @@
+  num_encoder_tokens, body_pp = load_text_processor(fname='body_pp.dpkl')
+
+  """
+  # Load files from disk
+  with open(fname, 'rb') as f:
+    pp = dpickle.load(f)
+
+  num_tokens = max(pp.id2token.keys()) + 1
+  print('Size of vocabulary for {}: {}'.format(fname, num_tokens))
+  return num_tokens, pp
+
+
+def load_decoder_inputs(decoder_np_vecs='train_title_vecs.npy'):
+  """
+  Load decoder inputs.
+
+  Parameters
+  ----------
+  decoder_np_vecs : str
+    filename of serialized numpy.array of decoder input (issue title)
+
+  Returns
+  -------
+  decoder_input_data : numpy.array
+    The data fed to the decoder as input during training for teacher forcing.
+    This is the same as `decoder_np_vecs` except the last position.
+  decoder_target_data : numpy.array
+    The data that the decoder data is trained to generate (issue title).
+    Calculated by sliding `decoder_np_vecs` one position forward.
+
+  """
+  vectorized_title = np.load(decoder_np_vecs)
+  # For Decoder Input, you don't need the last word as that is only for prediction
+  # when we are training using Teacher Forcing.
+  decoder_input_data = vectorized_title[:, :-1]
+
+  # Decoder Target Data Is Ahead By 1 Time Step From Decoder Input Data (Teacher Forcing)
+  decoder_target_data = vectorized_title[:, 1:]
+
+  print('Shape of decoder input: {}'.format(decoder_input_data.shape))
+  print('Shape of decoder target: {}'.format(decoder_target_data.shape))
+  return decoder_input_data, decoder_target_data
+
+
+def load_encoder_inputs(encoder_np_vecs='train_body_vecs.npy'):
+  """
+  Load variables & data that are inputs to encoder.
+
+  Parameters
+  ----------
+  encoder_np_vecs : str
+    filename of serialized numpy.array of encoder input (issue title)
+
+  Returns
+  -------
+  encoder_input_data : numpy.array
+    The issue body
+  doc_length : int
+    The standard document length of the input for the encoder after padding
+    the shape of this array will be (num_examples, doc_length)
+
+  """
+  vectorized_body = np.load(encoder_np_vecs)
+  # Encoder input is simply the body of the issue text
+  encoder_input_data = vectorized_body
+  doc_length = encoder_input_data.shape[1]
+  print('Shape of encoder input: {}'.format(encoder_input_data.shape))
+  return encoder_input_data, doc_length
+
+
+def viz_model_architecture(model):
+  """Visualize model architecture in Jupyter notebook."""
+  display(SVG(model_to_dot(model).create(prog='dot', format='svg')))
+
+
+def free_gpu_mem():
+  """Attempt to free gpu memory."""
+  K.get_session().close()
+  cfg = K.tf.ConfigProto()
+  cfg.gpu_options.allow_growth = True
+  K.set_session(K.tf.Session(config=cfg))
+
+
+def test_gpu():
+  """Run a toy computation task in tensorflow to test GPU."""
+  config = tf.ConfigProto()
+  config.gpu_options.allow_growth = True
+  session = tf.Session(config=config)
+  hello = tf.constant('Hello, TensorFlow!')
+  print(session.run(hello))
+
+
+def plot_model_training_history(history_object):
```

Comment:
Can we not use TensorBoard?

Reply:
I think this is leftover from notebook, I'll clean up these from code, seq2seq utils right now is just copy paste